### PR TITLE
Refresh CSRF tokens and retry when we have issues 

### PR DESCRIPTION
**Pull request by WilliamsJason**
If you reboot a device the CSRF token could change and you need to issue a GET before issuing your POST/PUT/DELETE.

Also, POST/PUT/DELETE can set a new CSRF token which we didn't account for previously.

### CODE REVIEW

**Comment by WilliamsJason** [comment]
Resolves Issue #166.

@GregPettyjohn , this should resolve the issue you've been seeing, but I tested it on a different OS version so let me know if that's not the case and I can drop my OS back and do more testing on that version.

File: WindowsDevicePortalWrapper/WindowsDevicePortalWrapper.UniversalWindows/HttpRest/RestDelete.cs [hunk 1]
```diff
@@ -56,9 +57,19 @@ public partial class DevicePortal
                 {
                     if (!response.IsSuccessStatusCode)
                     {
+                        // If this isn't a retry and it failed due to a bad CSRF
+                        // token, issue a GET to refresh the token and then retry.
+                        if (allowRetry && response.StatusCode == HttpStatusCode.Forbidden && response.ReasonPhrase.Equals("CSRF Token Invalid"))
+                        {
+                            await this.GetOperatingSystemInformation();
```
Comment (davidkline-ms) [inline WindowsDevicePortalWrapper/WindowsDevicePortalWrapper.UniversalWindows/HttpRest/RestDelete.cs]: I'd love to see something like:
await RefreshCsrfToken();

private async Task RefreshCsrfToken()
{
    await this.GetMachineName();
}

The idea is to as small of a request as we can and make it easy to update all locations at once if we find a better way to handle this.

File: GettingStarted.md [hunk 2]
```diff
@@ -24,6 +24,17 @@ portal.Connect().Wait(); // The ConnectionStatus event will tell you if it succe
 ```
 For complete examples and stepping off points see the SampleWdpClients.  A UWP version and WPF version of the sample app are included in the solution.
 
+### Verifying your connection
```
Comment (davidkline-ms) [inline GettingStarted.md]: Nicely written :)

File: WindowsDevicePortalWrapper/WindowsDevicePortalWrapper.UniversalWindows/HttpRest/RestDelete.cs (hunk 1 above)
Comment (WilliamsJason) [inline WindowsDevicePortalWrapper/WindowsDevicePortalWrapper.UniversalWindows/HttpRest/RestDelete.cs]: Sounds like a good idea.

**Pull request merged**